File: sdk/typescript/src/negotiation.ts
import { v7 as uuidv7 } from "uuid";
import type { AgentPayClient } from "./client";
import type { SLA } from "./types";

/**
 * Request a quote / SLA from a provider (MVP stub — R11.1).
 *
 * In MVP, the Negotiation_Engine is not yet implemented. This function
 * returns a fixed-template SLA with a 24-hour expiry. The Negotiation_Engine
 * (Task 14, post-MVP) will provide the full RFQ/quote/accept flow.
 *
 * @param client - The AgentPay API client
 * @param _providerHandle - Handle of the provider to request a quote from
 * @returns A stub SLA with a 24-hour expiry
 */
export async function requestQuote(
  client: AgentPayClient,
  _providerHandle: string,
): Promise<SLA> {
  return {
    slaId: uuidv7(),
    consumerHandle: "stub-consumer",
    providerHandle: _providerHandle,
    terms: {
      priceUsdcMicro: "0",
      description: "MVP stub — Negotiation_Engine not yet implemented",
    },
    expiresAt: new Date(Date.now() + 24 * 60 * 60 * 1000).toISOString(),
  };
}

File: sdk/python/src/agentpay_sdk/negotiation.py
"""Negotiation stub (mirrors sdk/typescript/src/negotiation.ts).

MVP: returns a fixed-template SLA with a 24-hour expiry.
Full RFQ/quote/accept flow arrives with Negotiation_Engine (Task 14, post-MVP).
"""

from uuid6 import uuid7
from datetime import datetime, timedelta, timezone

from .client import AgentPayClient
from .types import SLA


def request_quote(client: AgentPayClient, provider_handle: str) -> SLA:
    """Request a quote / SLA from a provider (MVP stub)."""
    return SLA(
        sla_id=str(uuid7()),
        consumer_handle="stub-consumer",
        provider_handle=provider_handle,
        terms={
            "priceUsdcMicro": "0",
            "description": "MVP stub — Negotiation_Engine not yet implemented",
        },
        expires_at=(datetime.now(timezone.utc) + timedelta(hours=24)).isoformat(),
    )
